src/ctapipe/atmosphere.py

```diff
@@ -59,30 +62,68 @@ def integral(self, height: u.Quantity) -> u.Quantity:
 
         """
 
-    def line_of_sight_integral(
-        self, distance: u.Quantity, zenith_angle=0 * u.deg, output_units=u.g / u.cm**2
+    @abc.abstractmethod
+    def height_from_overburden(self, overburden: u.Quantity) -> u.Quantity:
+        r"""Get the height a.s.l. from the mass overburden in the atmosphere.
+            Inverse of the integral function
+
+        Returns
+        -------
+        u.Quantity["m"]:
+            Height a.s.l. for given overburden
+
+        """
+
+    def slant_depth_from_height(
+        self,
+        height: u.Quantity,
+        zenith_angle=0 * u.deg,
+        output_units=GRAMMAGE_UNIT,
     ):
-        r"""Line-of-sight integral from the shower distance to infinity, along
+        r"""Line-of-sight integral from height to infinity, along
         the direction specified by the zenith angle. This is sometimes called
         the *slant depth*. The atmosphere here is assumed to be Cartesian, the
-        curvature of the Earth is not taken into account.
+        curvature of the Earth is not taken into account. Inverse of
+        height_from_slant_depth function.
```

Comment:
I'm fine with raising a warning here, and the docstring is easy of course. Should it be a custom warning type?

Reply:
I think you could create e.g.:
```py3
class SlantDepthZenithRangeWarning(UserWarning):
    """
    Issued when the zenith angle of an event is beyond
    the range where the slant depth computation is correct
    """ 

# then later:
warnings.warn("zenith angle too high for accurate slant depth", SlantDepthZenithRangeWarning)
```

I think the default warning filter is to issue the warning only once (as long as the message is static), so that would avoid the case where for very-large-zenith-angle observations it would trigger for many or all events (which would clutter the logs)

I also prefer a warning here to an exception the more I think about it, since there will be random events where this could happen due to mis-reconstruction, or as mentioned earlier for high zenith observations where the bottom of the FOV is below 70 deg.

Reply:
Really there is no much the user can do, other than to be aware of this and potentially filter events if they are relying on an accurate x_max... it should be a rare case, though

Reply:
Disappointed that it's not `FlatEarthWarning`

Reply:
Done